### PR TITLE
Allow arbitrary calls in the AST

### LINKED ISSUE #54: Allow calls with arbitrary expressions
The AST of Pijama only allows calling functions using their name.
```rust
enum Node<'a> {
    ...
    Call(Name<'a>, Block<'a>),
}
```
Which means that expressions like 
```elixir
(fn (x: Int) do x + 1 end)(1)
```
cannot be represented in the current AST. However, the IRs don't have any problem with that they can represent applications of arbitrary terms. 

The following changes need to be done to achieve this:
- [ ] Change the `ast::Node::Call` variant to `Node::Call(Node, Block)`
- [ ] Update the `parser::node::call` parser to allow either `Name`s or `Node`s inside brackets (this restriction exists just to prevent parsing ambiguities if any).
- [ ] Update the `mir::lower_call` function to handle `Node`s instead of `Name`s in its first argument.
- [ ] Add tests using the example above for:
    - [ ] Parsing.
    - [ ] Typing.
- [ ] Repurpose the `fancy_max` test to use its anonymous function directly.

### PR DESCRIPTION
**Pull request by pvdrz**
Fixes #54. This allows putting arbitrary expressions in the callee positions for function calls:
```elixir
(fn (x: Int) do x end) (1)
```
However such expressions are only allowed if they are inside brackets. This means that the following will not parse
```elixir
fn (x: Int) do x end (1)
```
This does not affect the previous behavior at all as confirmed by the old tests.

### CODE REVIEW

File: .rustfmt.toml [hunk 1]
```diff
@@ -1,5 +1,2 @@
 edition = "2018"
-format_code_in_doc_comments = true
-format_macro_matchers = true
-merge_imports = true
```
Comment (DarkDrek) [inline .rustfmt.toml]: Is this really necessary? There are warning in stable but we can make nightly a requirement for formatting / developing.

Comment (pvdrz) [reply .rustfmt.toml]: I'm not happy with it either but that's the price we have to pay for developing in stable.

I'm not sure if I like the idea of having nightly as a dev requirement. what if something runs in nightly but not in stable? of course CI will catch it but then you have to tell everyone to have nightly for formatting purposes and then switch to stable to run the tests locally and be sure the code will pass CI.

Comment (DarkDrek) [reply .rustfmt.toml]: > I'm not happy with it either but that's the price we have to pay for developing in stable.

I thought we just wanted it to compile / be usable on stable? And you can just use stable as default but for formatting use `cargo +nightly fmt`. I don't think it is that inconvenient.

Comment (pvdrz) [reply .rustfmt.toml]: I'll add a comment to the contrib file

**Pull request merged**